Source/WebKit/WebProcess/WebCoreSupport/WebPermissionController.cpp

```diff
@@ -26,33 +26,32 @@
 #include "config.h"
 #include "WebPermissionController.h"
 
-
 #include "WebPage.h"
+#include "WebPageProxyIdentifier.h"
 #include "WebPageProxyMessages.h"
+#include "WebPermissionControllerProxyMessages.h"
+#include "WebProcess.h"
+#include <WebCore/Page.h>
 #include <WebCore/PermissionObserver.h>
+#include <WebCore/PermissionState.h>
 
 namespace WebKit {
 
-Ref<WebPermissionController> WebPermissionController::create(WebPage& page)
+Ref<WebPermissionController> WebPermissionController::create()
 {
-    return adoptRef(*new WebPermissionController(page));
+    return adoptRef(*new WebPermissionController);
 }
 
-WebPermissionController::WebPermissionController(WebPage& page)
-    : m_page(page)
-{
-}
+WebPermissionController::WebPermissionController() = default;
 
-void WebPermissionController::query(WebCore::ClientOrigin&& origin, WebCore::PermissionDescriptor&& descriptor, CompletionHandler<void(std::optional<WebCore::PermissionState>)>&& completionHandler)
+void WebPermissionController::query(WebCore::ClientOrigin&& origin, WebCore::PermissionDescriptor&& descriptor, WebCore::Page& page, CompletionHandler<void(std::optional<WebCore::PermissionState>)>&& completionHandler)
 {
-    if (!m_page)
-        return completionHandler({ });
-
     auto cachedResult = queryCache(origin, descriptor);
     if (cachedResult != WebCore::PermissionState::Prompt)
         return completionHandler(cachedResult);
 
-    m_requests.append(PermissionRequest { WTFMove(origin), WTFMove(descriptor), WTFMove(completionHandler) });
+    auto pageIdentifier = WebPage::fromCorePage(page).identifier();
```

Comment:
Why are you storing a PageIdentifier when you really need a WebPageProxyIdentifier?

We would probably call webPageProxyIdentifier() here.

Reply:
We've gone back to using WebPageProxyIdentifier here. Fixed!